src/npm.js

```diff
@@ -27,14 +27,15 @@ class Workspace {
   */
   async extract(url, target) {
     debug('extract', { url, target });
-    let path = fsPath.join(this.dir, 'node_modules.tar.gz');
+    let path = fsPath.join(target, 'node_modules.tar.gz');
 
-    let req = download({ extract: true }).
+    let req = download().
       get(url).
       dest(target).
-      use(downloadStatus());
+      use(downloadStatus())
 
     await denodeify(req.run).call(req);
+    await run('tar', ['zxf', path], { cwd: this.dir });
```

Comment:
I added notes in the code and will add a new bug for this (under taskcluster optimizations) since we'll likely be able to get platform native compression (eg. tar.gz on linux and mac, .zip on windows) via a compression module or by writing it ourselves.